### DISCUSSION TITLE
Publish openapi module as separate package

### DISCUSSION THREAD

**Discussion by dbanty** [fastapi/fastapi#9042] [Questions]
In the same vein as #1107, it would be super useful (at least to me) to have the openapi models be available outside of FastAPI.  I've been working on a Python client generator (built with Typer!) which will hopefully be much cleaner than the openapi-generator version, and I already use FastAPI for some testing.  

I don't want to ship a copy of FastAPI with my tool just to get the openapi module, but it also feels silly to be rewriting a lot of what you already have when I could just parse a spec into your models and work from there.  So what I'll probably do is extract a copy of the openapi module and use it (with proper attribution of course), but it would be much cleaner to add your module as a pip dependency.

https://github.com/triaxtec/openapi-python-client is the project in question, just FYI.

**Comment by tiangolo** [comment]
That's awesome! :rocket: 

I had thought about that but never did it as there was no clear need yet.

I think it's probably a good idea. I will have to first update the models to use the new forward references instead of duplicating some models, but if that's useful for other tools now, I think it would make sense to do it. :rocket:
